### PR TITLE
Prac 06 code review request

### PR DESCRIPTION
**Pull request by Elsa-Stapelbroek**
Hi @callumjriley3 ! Would you be able to check the general structure of my code? I can't think of any specific sticking points, but feel like I'm forgetting something important. Thank you!

### CODE REVIEW

File: prac_06/used_cars.py [hunk 1]
```diff
@@ -9,10 +9,16 @@
...
     """Demo test code to show how to use car class."""
-    my_car = Car(180)
+    my_car = Car("My Car", 180)
     my_car.drive(30)
     print(f"Car has fuel: {my_car.fuel}")
     print(my_car)
 
+    limo = Car("Limo", 100)
+    limo.add_fuel(20)
+    print(limo.fuel)
```
Comment (callumjriley3) [inline prac_06/used_cars.py:19]: Could you format this print statement similar to the previous "Car has fuel: {my_car.fuel}"? Just a small suggestion!

File: prac_06/programming_language.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def __str__(self):
+        """Return object description in formatted string."""
+        return f"{self.name}, {self.typing} Typing, Reflection={self.reflection}, First appeared in {self.year}"
+
+    def is_dynamic(self):
+        """Determine whether the language uses dynamic or static typing."""
```
Comment (callumjriley3) [inline prac_06/programming_language.py:19]: Good docstrings for methods and class

File: prac_06/languages.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+"""
+Programming Languages
+estimated: 30 minutes
+actual: 16 minutes for main code, 22 including docstrings etc.
+"""
+from programming_language import ProgrammingLanguage
```
Comment (callumjriley3) [inline prac_06/languages.py]: Should this be:
from prac_06.programming_language import ProgrammingLanguage
?

File: prac_06/languages.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+python = ProgrammingLanguage("Python", "Dynamic", True, 1991)
+ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+print(python)
+
+languages = [python, ruby, visual_basic]
+print("The dynamically typed languages are:")
+print("\n".join(language.name for language in languages if language.is_dynamic()))
```
Comment (callumjriley3) [inline prac_06/languages.py:15]: Nice use of generator expression!

File: prac_06/guitar.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+"""Guitar class."""
+
+CURRENT_YEAR = 2023
+VINTAGE_MINIMUM = 50
```
Comment (callumjriley3) [inline prac_06/guitar.py]: Could this constant name be slightly more specific by including the word 'years'?

File: prac_06/guitar_test.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+"""Guitars part 2: Testing"""
+from guitar import Guitar
```
Comment (callumjriley3) [inline prac_06/guitar_test.py]: Should you specify that **guitar** is coming from the prac_06 folder?

File: prac_06/guitar_test.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+CURRENT_YEAR = 2023  # Doesn't affect program, maybe edit class for CURRENT_YEAR to be determined in client code?
+
+gibson = Guitar("Gibson L-5 CES", 1922, 16035.40)
+other = Guitar("another guitar", 2013)
+
+print(f"{gibson.name} get_age() - Expected {101}. Got {gibson.get_age()}")
```
Comment (callumjriley3) [inline prac_06/guitar_test.py:9]: Nice updating the expected output from the prac!

File: prac_06/guitars.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    guitars.append(guitar)
+    print(f"{guitar} added.")
+    name = input("Name: ")
+
+# for testing:
+# guitars.append(Guitar("Gibson L-5 CES", 1922, 16035.40))
+# guitars.append(Guitar("Line 6 JTV-59", 2010, 1512.9))
+
+for i, guitar in enumerate(guitars, 1):
+    vintage_string = " (vintage)" if guitar.is_vintage() else ""
```
Comment (callumjriley3) [inline prac_06/guitars.py:28]: Nice use of ternary operator!

File: prac_06/guitars.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
+"""
+Guitars part 3: get and display list of Guitars from user input
+
+Guitars combined time:
+estimate: 40 mins?
+actual: 50 mins
+"""
+from guitar import Guitar
```
Comment (callumjriley3) [inline prac_06/guitars.py]: (See previous suggestions)

**Comment by callumjriley3** [comment]
The general structure of your code seems correct and understandable. There's just a few minor suggestions that I made. :)

File: prac_06/languages.py (hunk 3 above)
Comment (Elsa-Stapelbroek) [reply prac_06/languages.py]: good point! Completely forgot about that

**Pull request merged**